### PR TITLE
Add SpikeMixture measure

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: src/combinators/spikemixture.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    w̄::S
+end
+SpikeMixture(m, w) = SpikeMixture(m, w, 1-w)
+
+function Base.show(io::IO, μ::SpikeMixture)
+    io = IOContext(io, :compact => true)
+    print(io, "(", μ.w, "*", string(μ.m), " + ", μ.w̄, "Dirac(0))")
+end
+
+basemeasure(μ::SpikeMixture) = SpikeMixture(basemeasure(μ.m), 1.0, (1/μ.w - 1)/density(μ.m, 0))
```
Comment (mschauer) [inline src/combinators/spikemixture.jl:15]: the second value is my `1/kappa`

File: src/combinators/spikemixture.jl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+function MeasureTheory.logdensity(μ::SpikeMixture, x)
+    return log(μ.w) + logdensity(μ.m, x)
+end
+function sampletype(μ::SpikeMixture)
+    sampletype(μ.m)
+end
+function Base.rand(rng::AbstractRNG, T::Type, μ::SpikeMixture)
+     return T((rand(rng) < μ.w)*rand(rng, μ.m))
```
Comment (mschauer) [inline src/combinators/spikemixture.jl:23]: Not so sure what to do in sampling if `μ.w + μ.w̄` don't add up to one.

**Comment by mschauer** [comment]
@sebagraz

**Review by cscherrer — COMMENTED** [review]
Thanks @mschauer , I've added a couple of comments. Could you also bump the version to 0.4.2?

File: src/combinators/spikemixture.jl (hunk 1 above)
Comment (mschauer) [reply src/combinators/spikemixture.jl]: We are working hard on making that possible end of this week ;-)

Comment (mschauer) [reply src/combinators/spikemixture.jl]: I mean adding a reference

**Review by cscherrer — APPROVED** [review]
@mschauer Looks like it needs to be 0.4.2, after that's bumped I think it's good whenever you're happy with it

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "MeasureTheory"
 uuid = "eadaa1a4-d27c-401d-8699-e962e1bbc33b"
 authors = ["Chad Scherrer <chad.scherrer@gmail.com> and contributors"]
-version = "0.4.0"
+version = "0.4.1"
```
Comment (mschauer) [inline Project.toml:4]:
```suggestion
version = "0.4.2"
```

**Pull request merged**